code/parachain/frame/cosmwasm/src/ibc.rs

```diff
@@ -394,22 +406,26 @@ impl<T: Config + Send + Sync> IbcModule for Router<T> {
 		let gas = Weight::MAX;
 		let mut vm = <Pallet<T>>::do_create_vm_shared(gas, InitialStorageMutability::ReadWrite);
 		let mut executor = Self::relayer_executor(&mut vm, address, contract_info).unwrap();
-		// let result = cosmwasm_call_serialize::<
-		// 	IbcChannelOpen,
-		// 	WasmiVM<CosmwasmVM<T>>,
-		// 	IbcPacketReceive,
-		// >(&mut executor, &message).unwrap();
-		// .map_err(|err| IbcError::implementation_specific(format!("{:?}", err)))
-		// .map(|x| match x.0 {
-		// 	ContractResult::Ok(version) => Ok(version),
-		// 	ContractResult::Err(err) =>
-		// 		Err(IbcError::implementation_specific(format!("{:?}", err))),
-		// })??
+		let result = cosmwasm_call_serialize::<
```

Comment:
On any operation other than `IbcChannelOpen`, `cosmwasm_system_entrypoint_serialize` must be called. Because there is orchestration happening. See https://github.com/ComposableFi/cosmwasm-vm/blob/master/vm-wasmi/src/semantic.rs#L1364-L1383

Reply:
Note that for the result, all IBC types are coerce back to `Response` because there is basically no need to have different types. Which mean that for `IbcPacketReceive` the response will contains a `data` instead of an `acknowledgement` field. The `data` field must be `Some(x)`.